ui-admin/src/study/kits/KitList.tsx

```diff
@@ -1,27 +1,88 @@
 import React, { useEffect, useState } from 'react'
-import _countBy from 'lodash/countBy'
-import { Link } from 'react-router-dom'
+import _fromPairs from 'lodash/fromPairs'
+import _groupBy from 'lodash/groupBy'
+import { Link, Navigate, NavLink, Route, Routes } from 'react-router-dom'
 import {
   ColumnDef,
-  getCoreRowModel, getFilteredRowModel, getSortedRowModel, useReactTable
+  getCoreRowModel, getFilteredRowModel, getSortedRowModel, SortingState, useReactTable, VisibilityState
 } from '@tanstack/react-table'
 
-import Api, { KitRequest } from 'api/api'
+import Api, { KitRequest, PepperKitStatus } from 'api/api'
 import { StudyEnvContextT } from 'study/StudyEnvironmentRouter'
 import LoadingSpinner from 'util/LoadingSpinner'
-import { basicTableLayout, FilterType } from 'util/tableUtils'
+import { basicTableLayout, ColumnVisibilityControl } from 'util/tableUtils'
 import { instantToDateString, isoToInstant } from 'util/timeUtils'
 
-const pepperStatusToHumanStatus = (dsmStatus?: string): string => {
-  const statusMap: Record<string, string> = {
-    'CREATED': 'Created',
-    'LABELED': 'Prepared',
-    'SCANNED': 'Sent',
-    'RECEIVED': 'Returned'
+type KitStatusTabConfig = {
+  status: string,
+  label: string,
+  additionalColumns?: string[]
+}
+
+const defaultColumns = {
+  'enrollee_shortcode': true,
+  'kitType_displayName': true,
+  'createdAt': true,
+  'pepperStatus_labelDate': false,
+  'pepperStatus_trackingNumber': false,
+  'pepperStatus_scanDate': false,
+  'pepperStatus_returnTrackingNumber': false,
+  'pepperStatus_receiveDate': false,
+  'pepperStatus': false
+}
+
+const statusTabs: KitStatusTabConfig[] = [
+  {
+    status: 'CREATED',
+    label: 'Created',
+    additionalColumns: []
+  },
+  {
+    status: 'LABELED',
+    label: 'Labeled',
+    additionalColumns: [
+      'pepperStatus_labelDate', 'pepperStatus_trackingNumber'
+    ]
+  },
+  {
+    status: 'SCANNED',
+    label: 'Sent',
+    additionalColumns: [
+      'pepperStatus_labelDate', 'pepperStatus_trackingNumber',
+      'pepperStatus_scanDate', 'pepperStatus_returnTrackingNumber'
+    ]
+  },
+  {
+    status: 'RECEIVED',
+    label: 'Returned',
+    additionalColumns: [
+      'pepperStatus_labelDate', 'pepperStatus_trackingNumber',
+      'pepperStatus_scanDate', 'pepperStatus_returnTrackingNumber',
+      'pepperStatus_receiveDate'
+    ]
+  },
+  {
+    status: 'ERROR',
+    label: 'Issues',
+    additionalColumns: [
+      'pepperStatus_labelDate', 'pepperStatus_trackingNumber',
+      'pepperStatus_scanDate', 'pepperStatus_returnTrackingNumber',
+      'pepperStatus_receiveDate',
+      'pepperStatus'
+    ]
+  }
+]
+
```

Comment:
a brief comment explaining what the intended visibility state is would be helpful